Sources/BSON/BSONError.swift

```diff
@@ -42,6 +42,19 @@ public enum BSONError {
     }
 }
 
+/// Standardize the errors emitted by BSONValue initializers.
+internal func BSONDecoderError(
```

Comment:
this function name makes it seem like it's a different error type or struct, rather than just a helper function. Instead, let's add it as an extension to `DecoderError` like we have for the other ones. e.g. `DecoderError.extendedJSONError(...)`.

Reply:
I changed it. followed the _ format from the other extensions. is that fine?